gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -357,6 +354,23 @@ public void onCompleted() {
       }
     }
 
+    private void runWithRetries(Runnable block, SimpleResponseObserver responseObserver)
+        throws IOException {
+      for (int attemptedRetries = 0; ; attemptedRetries++) {
+        block.run();
+        if (responseObserver.hasError()) {
+          if (attemptedRetries >= UPLOAD_RETRIES) {
```

Comment:
Move this condition in the `for` loop and throw exception outside the loop, and return from the method if there are no errors:
```java
    private void runWithRetries(Runnable block, SimpleResponseObserver responseObserver)
        throws IOException {
      for (int attemptedRetries = 0; attemptedRetries < UPLOAD_RETRIES; attemptedRetries++) {
        block.run();
        if (!responseObserver.hasError()) {
          return;
        }
      }
      throw new IOException(
          String.format("Failed to start resumable upload for '%s'", resourceId),
          responseObserver.getError());
    }
```

Reply:
Deleted.